src/deroff.rs

```diff
@@ -455,115 +454,694 @@ impl Deroffer {
+            } else {
+                return false;
+            }
+
+            if self.reg_table.contains_key(&reg) {
+                // This unwrap is safe because of the if
+                self.s = self.reg_table.get(&reg).unwrap().to_owned();
+                // This is unimplemented, so tests fail:
+                // self.text_arg();
+                return true;
+            } else {
+                return false;
+            }
+        } else {
+            return false;
+        }
+    }
+
+    fn size(&mut self) -> bool {
+        /* # We require that the string starts with \s */
+        if self.digit(2) || ("-+".contains(self.str_at(2)) && self.digit(3)) {
+            self.skip_char(3);
+            while self.digit(0) {
+                self.skip_char(1);
+            }
+            true
+        } else {
+            false
+        }
+    }
+
+    fn esc(&mut self) -> bool {
+        /* # We require that the string start with backslash */
+        // self.s.get(1..2).
+        match self.s.to_owned().get(1..2) {
+            Some(a) => match a {
+                "e" | "E" => self.condputs("\\"),
+                "t" => self.condputs("\t"),
+                "0" | "~" => self.condputs(" "),
+                "|" | "^" | "" | ":" => (),
+                o => self.condputs(o),
+            },
+            None => return false,
+        };
+        self.skip_char(2);
+        true
+    }
+
+    fn word(&mut self) -> bool {
+        let mut got_something = false;
+        while let Some(m) = self.g_re_word.find(&self.s.clone()) {
+            got_something = true;
+            self.condputs(m.as_str());
+            self.skip_char(m.end());
+
+            while self.spec() {
+                if !self.specletter {
+                    break;
+                }
+            }
+        }
+        got_something
+    }
+
+    fn text(&mut self) -> bool {
+        loop {
+            if let Some(idx) = self.s.clone().find("\\") {
+                self.condputs(self.s.clone().get(..idx).unwrap_or("")); // TODO: Fix! this may cause bugs later
+                self.skip_char(idx);
+                if !self.esc_char_backslash() {
+                    self.condputs(self.s.clone().get(0..1).unwrap_or("")); // TODO: Fix! this may cause bugs later
+                    self.skip_char(1);
+                }
+            } else {
+                self.condputs(&self.s.clone());
+                self.s = String::new();
+                break;
+            }
+        }
+        true
+    }
+
+    fn spec(&self) -> bool {
         unimplemented!()
     }
 
-    fn not_whitespace(s: &str, idx: usize) -> bool {
-        // # Note that this return False for the empty string (idx >= len(self.s))
-        // ch = self.s[idx:idx+1]
-        // return ch not in ' \t\n'
-        // TODO Investigate checking for ASCII whitespace after mvp
-        s.get(idx..(idx + 1))
-            .map(|string| " \t\n".contains(string))
-            .unwrap_or_default()
```

Comment:
Maybe you could add a comment
```rust
    /// Like esc_char, but we know the string starts with a backslash
```
Based on https://github.com/fish-shell/fish-shell/blob/master/share/tools/deroff.py#L613